sites/dotcom/document.js

```diff
@@ -15,8 +18,37 @@ type Props = {
 
 export default ({ Page, data: { body, ...data } }: Props) => {
     const cleanedData = { ...parseCAPI(body), ...data };
-    return document({
-        Page: () => <App data={{ ...cleanedData }} Page={Page} />,
-        data: cleanedData,
+    const bundle = [
+        'javascript',
+        `${cleanedData.site}.${cleanedData.page.toLowerCase()}.js`,
+    ];
+    const { html, css, ids: cssIDs } = renderToString(
+        <App data={{ ...cleanedData }} Page={Page} />,
+    );
+    /**
+     * To save sending CAPI data twice (in the HEAD and BODY)
+     * we replace any keys present in the CapiComponent keyRegister
+     * with document.querySelectors, therefore sending the data in the BODY only
+     * */
+    const CAPI = Object.keys(cleanedData.CAPI).reduce((acc, key) => {
```

Comment:
i wonder if we should move this lookup stuff into the CAPIcomponent itself? if it cannot find an element with correct `data-` attr to grab the innerHTML from, then pull it from the global state? then this would just need to delete the keys from the config obj.

it feels cleaner and less leaky that way, but was there also a reason not to i've forgotten? i have a suspicion there is...

Reply:
By deleting the keys from the config object we'd basically be saying this CAPI data isn't going to be available to any other component which doesn't happen to be a `CapiComponent`. I don't know how likely that is to happen? That we'll have 2 components on a page using the same CAPI data, one being a `CapiComponent` and the other being a "dynamic" component.